activesupport/lib/active_support/core_ext/symbol/inflections.rb

```diff
@@ -0,0 +1,126 @@
+# frozen_string_literal: true
+
+require "active_support/code_generator"
+require "active_support/core_ext/string/inflections"
+
+# Symbol inflections define new methods on the Symbol class to transform names for different purposes.
+# For instance, you can figure out the name of a table from the name of a class.
+#
+#   :ScaleScore.tableize # => :scale_scores
+#
+class Symbol
+  ##
+  # :method: camelize
+  # :call-seq: camelize(first_letter = :upper)
+  #
+  # See String#camelize.
+
+  ##
+  # :method: classify
+  #
+  # See String#classify.
+
+  ##
+  # :method: dasherize
+  #
+  # See String#dasherize.
+
+  ##
+  # :method: deconstantize
+  #
+  # See String#deconstantize.
+
+  ##
+  # :method: demodulize
+  #
+  # See String#demodulize.
+
+  ##
+  # :method: downcase_first
+  #
+  # See String#downcase_first.
+
+  ##
+  # :method: foreign_key
+  # :call-seq: foreign_key(separate_class_name_and_id_with_underscore = true)
+  #
+  # See String#foreign_key.
+
+  ##
+  # :method: humanize
+  # :call-seq: humanize(capitalize: true, keep_id_suffix: false)
+  #
+  # See String#humanize.
+
+  ##
+  # :method: parameterize
+  # :call-seq: parameterize(separator: "-", preserve_case: false, locale: nil)
+  #
+  # See String#parameterize.
+  #
+  # Raises EncodingError if the Symbol#name contains invalid UTF-8
+
+  ##
+  # :method: pluralize
+  # :call-seq: pluralize(count = nil, locale = :en)
+  #
+  # See String#pluralize.
+
+  ##
+  # :method: singularize
+  # :call-seq: singularize(locale = :en)
+  #
+  # See String#singularize.
+
+  ##
+  # :method: tableize
+  #
+  # See String#tableize.
+
+  ##
+  # :method: titleize
+  # :call-seq: titleize(keep_id_suffix: false)
+  #
+  # See String#titleize
+
+  ##
+  # :method: underscore
+  #
+  # See String#underscore.
+
+  ##
+  # :method: upcase_first
+  #
+  # See String#upcase_first.
+
+  ActiveSupport::CodeGenerator.batch(self, __FILE__, __LINE__) do |owner|
+    [
+      :camelize,
+      :classify,
+      :dasherize,
+      :deconstantize,
+      :demodulize,
+      :downcase_first,
+      :foreign_key,
+      :humanize,
+      :parameterize,
+      :pluralize,
+      :singularize,
+      :tableize,
+      :titleize,
+      :underscore,
+      :upcase_first
+    ].each do |method_name|
+      owner.class_eval do |batch|
```

Comment:
I'm not sold on the usefulness of this feature, but this meta programming is very unnecessary, specially since we are already defining the "method" just to generate the documentation.

Reply:
Thank you for that feedback. You're right, metaprogramming is not necessary here. I've  unrolled this loop to instead explicitly define and document the methods sequentially.